### PR TITLE
DON-NOT-MERGE-RL-177 Read data from screening relationship endpoint - Part1 (Redux)

### PR DESCRIPTION
**Pull request by karthik-sista**
### Jira Story

- [Read Data from Screening Relationship Endpoint (FE-Redux React)](https://osi-cwds.atlassian.net/secure/RapidBoard.jspa?rapidView=180&projectKey=RL&modal=detail&selectedIssue=RL-177)

## Description
This story should be merged along with RL-151 i.e Part 2 (ruby) https://osi-cwds.atlassian.net/secure/RapidBoard.jspa?rapidView=180&projectKey=RL&modal=detail&selectedIssue=RL-151
The story is to the fetch relationships from the new endpoint (GET /screenings/{screeningId}/relationships_with_candidates) using **screeningId** . 

## Tests
- [x] I have included unit tests 
- [ ] I have included feature tests 
- [ ] I have included other tests 
- [ ] I have NOT included tests 

## Types of changes

- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] I have ran all tests for the project.
- [x] I have ran lint/rubocop check for the changed/new files.
- [x] My code is in a stable state ready to be deployable, but not necessarily complete.
- [x] I promise on my honor as a CWDS developer to ensure I don't break the build, to check Lint/Rubocop, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

File: app/javascript/sagas/createParticipantSaga.js [hunk 1]
```diff
@@ -43,10 +43,10 @@ export function* createParticipant({payload: {person}}) {
   try {
     const response = yield* sendPersonPayload(person)
     yield put(createPersonSuccess(response))
-    const clientIds = yield select(selectClientIds)
-    yield put(fetchRelationships(clientIds))
     const screeningId = yield select(getScreeningIdValueSelector)
     yield put(fetchHistoryOfInvolvements('screenings', screeningId))
+    const clientIds = yield select(selectClientIds)
+    yield put(fetchRelationships(clientIds, screeningId))
```
Comment (wmitchell-cambria) [inline app/javascript/sagas/createParticipantSaga.js]: I know we talked about moving the client select, that seems fine. I'm curious if there was a reason to move the `fetchRelationships` to be after `fetchHOI`. They probably both run in parallel in practice, but it seems odd loading them in reverse order from how they are displayed on the page.

File: app/javascript/sagas/fetchRelationshipsSaga.js [hunk 2]
```diff
@@ -8,9 +8,9 @@ import {
   FETCH_RELATIONSHIPS,
 } from 'actions/actionTypes'
 
-export function* fetchRelationships({payload: {ids}}) {
+export function* fetchRelationships({payload: {ids, screeningId}}) {
   try {
-    const response = yield call(get, `/api/v1/relationships?clientIds=${ids.join(',')}`)
+    const response = yield call(get, `/api/v1/relationships?clientIds=${ids.join(',')}&screeningId=${screeningId}`)
```
Comment (wmitchell-cambria) [inline app/javascript/sagas/fetchRelationshipsSaga.js]: Are we adding this even when it is null? Is it possible to instead not add the parameter in that case?

File: app/javascript/sagas/createParticipantSaga.js (hunk 1 above)
Comment (karthik-sista) [reply app/javascript/sagas/createParticipantSaga.js]: There was no reason behind it. Ya makes sense to load in same order as the page. 
👍

File: app/javascript/sagas/fetchRelationshipsSaga.js (hunk 2 above)
Comment (amanbop) [reply app/javascript/sagas/fetchRelationshipsSaga.js]: We can do that way also. But we are checking in the api/controller if screeningId is null. And send the request to appropriate end point accordingly.

Comment (wmitchell-cambria) [reply app/javascript/sagas/fetchRelationshipsSaga.js]: 👍 kk, either way seems fine in the end.

**Comment by wmitchell-cambria** [comment]
For any confused onlookers like me:

The title of this says 'Do Not Merge' but it was merged. Don't worry! This PR was not merged directly. Instead, all of its commits were included in #826, which was thoroughly reviewed and merged. GitHub just detected that these commits were included, and helpfully updated the status here.

**Pull request merged**